server/events_controller.go

```diff
@@ -552,3 +572,27 @@ func (e *EventsController) commentNotAllowlisted(baseRepo models.Repo, pullNum i
 		e.Logger.Err("unable to comment on pull request: %s", err)
 	}
 }
+
+// commentUserDoesNotHavePermissions comments on the pull request that the user
+// is not allowed to execute the command.
+func (e *EventsController) commentUserDoesNotHavePermissions(baseRepo models.Repo, pullNum int, user models.User, cmd *events.CommentCommand) {
+	errMsg := fmt.Sprintf("```\nError: User @%s does not have permissions to execute '%s' command.\n```", user.Username, cmd.Name)
+	if err := e.VCSClient.CreateComment(baseRepo, pullNum, errMsg, ""); err != nil {
+		e.Logger.Err("unable to comment on pull request: %s", err)
+	}
+}
+
+// checkUserPermissions checks if the user has permissions to execute the command
+func (e *EventsController) checkUserPermissions(repo models.Repo, user models.User, cmd *events.CommentCommand) (bool, error) {
+	if cmd.Name == models.ApplyCommand || cmd.Name == models.PlanCommand {
```

Comment:
Do we really need this check? Seems a bit redundant

Reply:
if we add more command like unlock, it could be parsed by this block which it will not work but it could be removed I think since at that point we know is a valid command

Reply:
I think by the time we get here the command has been parsed and it is valid. Also this might introduce bugs if there is a new command and this place was not updated. I believe this feature should work with any valid Atlantis command, right?